rspace/src/test/scala/coop/rchain/rspace/test/ArbitraryInstances.scala

```diff
@@ -84,6 +87,26 @@ object ArbitraryInstances {
         .map(_.toMap))
   }
 
+  implicit val arbitraryWaitingContinuation
+    : Arbitrary[WaitingContinuation[Pattern, StringsCaptor]] = {
+    Arbitrary(Gen.sized { size =>
+      val constrainedSize = if (size > 1) size else 1
+      for {
+        pats    <- Gen.containerOfN[List, Pattern](constrainedSize, Arbitrary.arbitrary[Pattern])
+        boolean <- Arbitrary.arbitrary[Boolean]
+      } yield (WaitingContinuation(pats, new StringsCaptor, boolean))
+    })
+  }
+
+  implicit val arbitraryGnat: Arbitrary[GNAT[String, Pattern, String, StringsCaptor]] = {
+    Arbitrary(for {
+      chans <- Gen.containerOf[List, String](Arbitrary.arbitrary[String])
+      data  <- Gen.containerOf[List, Datum[String]](Arbitrary.arbitrary[Datum[String]])
+      wks <- Gen.containerOf[List, WaitingContinuation[Pattern, StringsCaptor]](
+              Arbitrary.arbitrary[WaitingContinuation[Pattern, StringsCaptor]])
```

Comment:
It would be good defensively write this in a way to ensure that the `patterns.length` for each member of `wks` was the same as `chans.length`.  I've suggested something similar below.

Reply:
Done